src/App.tsx

```diff
@@ -16,47 +19,103 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goodsArr, setGoodsArr] = useState(goodsFromServer);
+  const [sortValue, setSortValue] = useState('');
+  const [reverseValue, setReverseValue] = useState(false);
+  const [resetValue, setResetValue] = useState(false);
+
+  const sortBy = (value: string) => {
+    if (value === variables.sort_by_alphabetically) {
+      if (reverseValue) {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => b.localeCompare(a)));
+      } else {
+        setGoodsArr(prevState => [...prevState].sort((a, b) => a.localeCompare(b)));
+      }
+
+      setSortValue(variables.sort_by_alphabetically);
+      setResetValue(true);
+
+      return;
+    }
+
+    if (reverseValue) {
+      setGoodsArr(prevState => [...prevState].sort((a, b) => b.length - a.length));
+    } else {
+      setGoodsArr(prevState => [...prevState].sort((a, b) => a.length - b.length));
+    }
+
+    setSortValue(variables.sort_by_length);
+    setResetValue(true);
+  };
+
+  const reset = () => {
+    setGoodsArr(goodsFromServer);
+    setReverseValue(false);
+    setResetValue(false);
+    setSortValue('');
+  };
+
+  const reverse = () => {
+    if (!sortValue) {
+      setResetValue(!resetValue);
+    } else {
+      setResetValue(true);
+    }
+
+    setGoodsArr(prevState => [...prevState].reverse());
+    setReverseValue(!reverseValue);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortValue !== variables.sort_by_alphabetically,
+          })}
+          onClick={() => sortBy(variables.sort_by_alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortValue !== variables.sort_by_length,
+          })}
+          onClick={() => sortBy(variables.sort_by_length)}
```

Comment:
```suggestion
          onClick={() => sortSortValue(SORT_BY.length)}
```
click on sort button should only set proper sort value in state.

Reply:
Вийшла якась каша, дайте більш зрозуміліший коментар

Reply:
if something doesn't work out, then please ask you questions in a fe_pt chat